### PR TITLE
[RAPTOR-6119] disallow Y transform

### PR DESCRIPTION
**Pull request by scottp-ml**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
At this time we do not want to allow users to transform the target value in transform tasks.  There is insufficient validation performed on those values and they are likely to cause issues in various features in DR such as insights.

### CODE REVIEW

File: custom_model_runner/datarobot_drum/drum/model_adapter.py [hunk 1]
```diff
@@ -403,10 +403,11 @@ def transform(self, model=None, **kwargs):
...
                 if type(transform_out) == tuple:
-                    output_data, output_target = transform_out
-                else:
-                    output_data = transform_out
-                    output_target = target_data
+                    raise DrumTransformException(
+                        "Transformation of the target variable is not supported by DRUM."
+                    )
+                output_data = transform_out
+                output_target = target_data
```
Comment (jmbannon) [inline custom_model_runner/datarobot_drum/drum/model_adapter.py:410]: We need this same logic in `score.R`

**Review by rvorobii — APPROVED** [review]

File: tests/drum/test_fit.py [hunk 2]
```diff
@@ -369,6 +367,33 @@ def test_transform_fit_fails_default(
             )
             assert "WARNING: No type schema provided. For transforms, we" not in stdout
 
+    @pytest.mark.parametrize(
+        "framework, language", [(SKLEARN_TRANSFORM_WITH_Y, PYTHON), (R_TRANSFORM_NO_Y, R_FIT),]
```
Comment (jmbannon) [inline tests/drum/test_fit.py:371]: R_TRANSFORM_NO_Y should fail this test, no y is passed back in it

File: tests/drum/test_fit.py [hunk 3]
```diff
@@ -369,6 +367,33 @@ def test_transform_fit_fails_default(
...
+            target_type,
+            custom_model_dir,
+            input_dataset,
+            "--disable-strict-validation",
+        )
+        with pytest.raises(AssertionError):
+            _, stdout, _ = _exec_shell_cmd(
+                cmd, "Failed in {} command line! {}".format(ArgumentsOptions.MAIN_COMMAND, cmd)
+            )
+            assert "Transformation of the target variable is not supported by DRUM." in stdout
```
Comment (jmbannon) [inline tests/drum/test_fit.py:395]: This assertion will not be reached. You need to add `assert_if_fail=False` to the _exec call above

File: tests/drum/test_inference.py [hunk 4]
```diff
@@ -376,15 +376,18 @@ def test_custom_models_with_drum_nginx_prediction_server(
...
-            (SKLEARN_TRANSFORM_DENSE, TRANSFORM, PYTHON_TRANSFORM_DENSE, None, True),
-            (SKLEARN_TRANSFORM, TRANSFORM, PYTHON_TRANSFORM, None, False),
-            (SKLEARN_TRANSFORM_DENSE, TRANSFORM, PYTHON_TRANSFORM_DENSE, None, False),
+            # The following 3 tests produce Y transform values and are being temporarily removed until y transform
+            # validation is added
+            # (SKLEARN_TRANSFORM_DENSE, TRANSFORM, PYTHON_TRANSFORM_DENSE, None, True),
+            # (SKLEARN_TRANSFORM, TRANSFORM, PYTHON_TRANSFORM, None, False),
+            # (SKLEARN_TRANSFORM_DENSE, TRANSFORM, PYTHON_TRANSFORM_DENSE, None, False),
             (SKLEARN_TRANSFORM, TRANSFORM, PYTHON_TRANSFORM_NO_Y, None, True),
+            (SKLEARN_TRANSFORM, TRANSFORM, PYTHON_TRANSFORM_NO_Y, None, False),
```
Comment (jmbannon) [inline tests/drum/test_inference.py:385]: nit: we should refactor these constants as follows:
PYTHON/R_TRANSFORM -> implies there is no y
PYTHON/R_TRANSFORM_WITH_Y -> implies there is y

Comment (scottp-ml) [reply tests/drum/test_inference.py:385]: I will work on refactoring that in a follow up https://datarobot.atlassian.net/browse/RAPTOR-6233

File: tests/drum/test_fit.py [hunk 5]
```diff
@@ -371,6 +369,32 @@ def test_transform_fit_fails_default(
             )
             assert "WARNING: No type schema provided. For transforms, we" not in stdout
 
+    @pytest.mark.parametrize("framework, language", [(SKLEARN_TRANSFORM_WITH_Y, PYTHON),])
+    @pytest.mark.parametrize("problem", [REGRESSION, BINARY, ANOMALY])
+    def test_transform_fit_disallow_y_output(
```
Comment (jmbannon) [inline tests/drum/test_fit.py:374]: We should add an R test here

**Pull request merged**